src/sortStudents.ts

```diff
@@ -1,16 +1,65 @@
+// import { arrayExpression } from '@babel/types';
 
 export interface Student {
   // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
   // describe SortType enum
+  Name = 'NAME',
+  Surname = 'SURNAME',
+  Age = 'AGE',
+  Married = 'MARRIED',
+  AverageGrade = 'AVERAGEGRADE',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
-
-export function sortStudents(students, sortBy, order) {
+export function sortStudents(students: Student[],
+  sortBy: SortType,
+  order: SortOrder)
+  : Student[] {
   // write your function
+  const result = [...students];
+
+  const averageGrade = (array:
+  number[]): number => (array.reduce((a, b) => a + b) / array.length);
+
+  result.sort((person1, person2) => {
+    let a = person1;
+    let b = person2;
+
+    if (order === 'desc') {
+      a = person2;
+      b = person1;
+    }
+
+    switch (sortBy) {
+      case SortType.Name:
+        return a.name.localeCompare(b.name);
+
+      case SortType.Surname:
+        return a.surname.localeCompare(b.surname);
```

Comment:
Combine similar data types sort
```suggestion
      case SortType.Name:
      case SortType.Surname:
        return a[sortBy].localeCompare(b[sortBy]);
```